java/client/src/main/java/glide/api/commands/GeospatialIndicesBaseCommands.java

```diff
@@ -135,4 +139,270 @@ CompletableFuture<Long> geoadd(
+     * Object[] result =
+     *                client
+     *                        .geosearch(
+     *                                "a1",
+     *                                new GeoSearchOrigin("Palermo"),
+     *                                new GeoSearchShape(200, GeoUnit.KILOMETERS))
+     *                        .get();
+     * assert expected.equals(result);
+     * }</pre>
+     */
+    CompletableFuture<String[]> geosearch(
+            String key, GeoSearchOrigin.SearchOrigin searchFrom, GeoSearchShape searchBy);
+
+    /**
+     * Returns the members of a sorted set populated with geospatial information using {@link
+     * #geoadd(String, Map)}, which are within the borders of the area specified by a given shape.
+     *
+     * @since Redis 6.2.0 and above.
+     * @see <a href="https://valkey.io/commands/geosearch">valkey.io</a> for more details.
+     * @param key The key of the sorted set.
+     * @param searchFrom The query's center point options:
+     *     <ul>
+     *       <li>MemberOrigin: Use the position of the given existing member in the sorted set.
+     *       <li>Use the given longitude and latitude position.
+     *     </ul>
+     *
+     * @param searchBy The query's shape options:
+     *     <ul>
+     *       <li>BYRADIUS: Search inside circular area according to given radius.
+     *       <li>BYBOX: Search inside an axis-aligned rectangle, determined by height and width.
+     *     </ul>
+     *
+     * @param resultOptions Contains up to 2 optional input:
+     *     <ul>
+     *       <li>SortOrder: The query's order to sort the results by:
+     *           <ul>
+     *             <li>ASC: Sort returned items from the nearest to the farthest, relative to the
+     *                 center point.
+     *             <li>DESC: Sort returned items from the farthest to the nearest, relative to the
+     *                 center point.
+     *           </ul>
+     *       <li>COUNT: Limits the results to the first N matching items, when the 'ANY' option is
+     *           used, the command returns as soon as enough matches are found. This means that the
+     *           results might may not be the ones closest to the origin.
+     *     </ul>
+     *
+     * @return An array of matched member names.
+     * @example
+     *     <pre>{@code
+     * Object[] expected = new String[] {"Catania", "Palermo"};
+     * Object[] result =
+     *                client
+     *                        .geosearch(
+     *                                "a1",
+     *                                new GeoSearchOrigin("Palermo"),
+     *                                new GeoSearchShape(200, GeoUnit.KILOMETERS),
+     *                                SortOrder.ASC)
+     *                        .get();
+     * assert expected.equals(result);
+     * }</pre>
+     */
+    CompletableFuture<String[]> geosearch(
+            String key,
+            GeoSearchOrigin.SearchOrigin searchFrom,
```

Comment:
You can add import and
```suggestion
            SearchOrigin searchFrom,
```